projects/storefrontlib/shared/models/file.ts

```diff
@@ -0,0 +1,21 @@
+export interface ExportFileOptions {
+  /**
+   * File name for exported file.
+   */
+  fileName: string;
+  /**
+   * Extension for exported file.
+   */
+  extension: string;
+  /**
+   * Mime/type for exported file.
+   */
+  type: string;
+}
+
+export interface CsvFileValidationErrors {
+  tooLarge?: { maxSize: number };
+  tooManyEntries?: { maxEntries: number };
+  empty?: boolean;
+  notParsable?: boolean;
+}
```

Comment:
nitpick: IMHO CsvFileValidationErrors doesn't belong to the shared file models. Can we move it near the csv related service?

Reply:
Where exactly? Into the service file/folder? I don't see a better place for that than now. BTW I guess we should still have it in public API.

Reply:
Agreed on `CsvFileValidationErrors` in public api.

Regarding the structure of storefrontlib, now I understand it looks like: 
```
projects/storefrontlib/shared/
├ services/
│  └ import-export/
│    ├ file-download.service.ts
│    ├ file-reader.service.ts
│    ├ import-csv-file.service.ts
│    └ export-csv-file.service.ts
└ models/
   └ file.ts (with CsvFileValidationErrors and ExportFileOptions)
``` 

What do you think about grouping file-related stuff in `/services/file/` folder?
And nesting the csv-specific stuff in `/services/file/csv` folder?
The csv file validation errors model could live in the `.../csv` folder.

What do you think about the following?:
```
projects/storefrontlib/shared/services/
└ file/
   ├ file-download.service.ts
   ├ file-reader.service.ts
   ├ file-export-options.ts (with FileExportOptions)
   └ csv/
     ├ import-csv-file.service.ts
     ├ export-csv-file.service.ts
     └ csv-file-validation-errors.ts  (with CsvFileValidationErrors)

```

Reply:
Ok, it looks clean for me! Thanks :)